express-notes-api/index.js

```diff
@@ -0,0 +1,100 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+const jsonMethod = express.json();
+app.use(jsonMethod);
+
+const data = require('./data.json');
+
+app.get('/api/notes', (req, res) => {
+  const dataArray = [];
+  for (const key in data.notes) {
+    if (data.notes[key]) {
+      dataArray.push(data.notes[key]);
+    }
+  }
+  res.status(200).json(dataArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const iD = req.params.id;
+  if (data.notes[iD]) {
+    if ((Math.sign(iD) === 1) && Number.isInteger(parseInt(iD)) && iD !== '0') {
+      res.status(200).send(data.notes[iD]);
+    }
+    res.status(404).send({ error: `cannot find note with id ${iD}` });
+  }
+  res.status(400).send({ error: 'id must be a positive integer' });
```

Comment:
Calling `res.send(...)` doesn't end the middleware function, so it's possible that `send` will be called three times on one request. Please make sure there is exactly one response sent for any given request.